src/BlockTypes/Cart.php

```diff
@@ -78,6 +78,20 @@ public function render( $attributes = array(), $content = '' ) {
 			$this->block_name . '-frontend',
 			$this->block_name . '-block-frontend'
 		);
+
+		// Disable storefront footer bar when using this block.
+		if ( has_action( 'storefront_footer', 'storefront_handheld_footer_bar' ) ) {
+			remove_action( 'storefront_footer', 'storefront_handheld_footer_bar', 999 );
+		}
```

Comment:
I'm wary that we're establishing a precedent here for handling theme specific customizations in the block plugin. What if we encounter other themes in the wild that add a similar mobile footer?

Some possible options:

a.  Leave this for now, deal with another approach if/when we find other themes that need dealt with.
b.  Create a theme compat file/class so we keep all theme specific compatibility things separate so as themes account for them we can remove the compat.  This might warrant adding action hooks to the blocks for hooking this compat file in.
c.  Put the compat code in the storefront theme (establishing the precedent that _themes_ have to account for the blocks).

I personally lean to b or c here. Thoughts?

Reply:
So the reason I put it here instead of directly in Storefront was, this action needs to run only when the block is used. Storefront has no way of knowing this, unless we add an action or something here for it to listen out for.

Alternatively we can just go with z-index for now. It's hacky but it will cover the Storefront footer.

Reply:
> Storefront has no way of knowing this, unless we add an action or something here for it to listen out for.

I think this may be a good signal of the need to _add_ an action here. In #1961 I added a set of actions to both the cart block and the checkout block that would work well for this.  Although we may want to look at automatically adding dynamic actions  on the via the render method in the abstract class so that every child block type already has them in place.

Reply:
Yes those 2 actions would be enough. Ill hold off for those to merge - we can raise PR in storefront.

Reply:
In this case I think a temporary hook on `storefront_footer` is ok, since it's our default theme, and these blocks are still in development.

Long term adding hooks for blocks and implementing these kinds of fixes in themes is the best approach, happy if we go that approach now. Related, we may need to fire certain existing cart/checkout hooks to ensure things like `woocommerceanalytics` events fire for block-based cart/checkout - e.g. [`woocommerce_after_checkout_form`](https://github.com/Automattic/jetpack/blob/master/modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php#L46). I'm guessing there are other hooks which we'll need to review/fire when cart & checkout blocks are fully functional.